src/game/AI/ScriptDevAI/scripts/outland/hellfire_citadel/shattered_halls/shattered_halls.h

```diff
@@ -72,8 +72,37 @@ enum
     NPC_HEARTHEN_GUARD          = 17621,
     NPC_SHARPSHOOTER_GUARD      = 17622,
     NPC_REAVER_GUARD            = 17623,
-
-    WORLD_STATE_CUSTOM_SPAWN_WAVES = 5400004,
+    
+    // First Group in Dungeon should not drop any loot/give any rep
+    NPC_SHATTERED_HAND_HEATHEN      = 17420,
+    NPC_SHATTERED_HAND_SAVAGE       = 16523,
+    STRING_ID_ENTRANCE_GROUP        = 5400002,
+    
+    // Used for Hall of Fathers intro
+    WORLD_STATE_LEGIONNAIRE_001     = 5400001,              // Prevents that shattered hand savage/heathen can respawn when legionnaire 01 is dead
+    WORLD_STATE_LEGIONNAIRE_002     = 5400002,              // Spawns Legionnaire 03 group
+    WORLD_STATE_LEGIONNAIRE_003     = 5400003,              // Spawns Legionnaire 04 and 05 groups
+    WORLD_STATE_CUSTOM_SPAWN_WAVES  = 5400004,              // Spawns initial waves at gauntlet of flame
+
+    // Legionnaire StringID
+    FIRST_LEGIONNAIRE_STRING        = 5400001,
```

Comment:
You do not need to specify string IDs by ID itself. Instead, just use the string version of it in place and then DB ID can change while string part stays the same. Thats why it works like this.